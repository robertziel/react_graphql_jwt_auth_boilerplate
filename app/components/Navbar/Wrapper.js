/* eslint-disable indent */

import styled from 'styled-components';

const Wrapper = styled.nav`
  background: #fafbfc;
  box-shadow: 0 10px 35px rgba(4, 9, 20, 0.04),
    -18px -30px 23px rgba(4, 9, 20, 0.04), 0 1px 8px rgba(4, 9, 20, 0.06),
    0 2px 2px rgba(4, 9, 20, 0.04);
  height: 60px;
  position: fixed;
  top: 0;
  right: 0px;
  width: ${({ marginLeft }) =>
    marginLeft ? `calc(100% - ${marginLeft}px)` : '100%'};

  .navbar-right {
    float: right;
    width: 60px;
    height: 60px;
    margin-right: 10px;
  }
`;

export default Wrapper;
