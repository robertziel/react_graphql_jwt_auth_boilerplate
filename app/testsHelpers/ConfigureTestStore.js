import { memoryHistory } from 'react-router-dom';
import configureStore from 'configureStore';
import StoreAccessor from 'containers/BackendApiConnector/StoreAccessor';

export default function ConfigureTestStore() {
  const confuguredStore = configureStore({}, memoryHistory);

  StoreAccessor.store = confuguredStore.store;

  this.store = confuguredStore.store;
  this.presistor = confuguredStore.presistor;
}
