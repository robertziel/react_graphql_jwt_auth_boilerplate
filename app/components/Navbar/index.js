import React from 'react';

import LanguageToggle from 'components/LanguageToggle/index';
import SignOutButton from 'components/SignOutButton/index';

import Wrapper from './Wrapper';

function Navbar() {
  return (
    <Wrapper>
      <div className="navbar-right">
        <SignOutButton />
      </div>
      <div className="navbar-right">
        <LanguageToggle />
      </div>
    </Wrapper>
  );
}

export default Navbar;
