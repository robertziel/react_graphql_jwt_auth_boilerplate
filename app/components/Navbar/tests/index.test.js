import React from 'react';
import { IntlProvider } from 'react-intl';
import { Provider } from 'react-redux';

import { mount } from 'enzyme';
import ConfigureTestStore from 'testsHelpers/ConfigureTestStore';

import Navbar from '../index';

// Mock LanguageToggle required by Navbar
/* eslint-disable react/prop-types */
jest.mock('components/LanguageToggle/index', () => () => (
  <div>LanguageToggle</div>
));
/* eslint-enable react/prop-types */

describe('<Navbar />', () => {
  it('should render and match the snapshot', () => {
    const { store } = new ConfigureTestStore();

    const wrapper = mount(
      <IntlProvider locale="en">
        <Provider store={store}>
          <Navbar />
        </Provider>
      </IntlProvider>,
    );
    expect(wrapper.html()).toMatchSnapshot();
  });
});
