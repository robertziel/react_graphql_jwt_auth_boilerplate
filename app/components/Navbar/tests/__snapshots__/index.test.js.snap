// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Navbar /> should render and match the snapshot 1`] = `"<nav class=\\"Wrapper-rw3tpv-0 ermAON\\"><div class=\\"navbar-right\\"><div class=\\"Wrapper-vnjem8-0 gCpCbH\\"><form><button class=\\"MuiButtonBase-root MuiButton-root MuiButton-text Button-sc-17yegtu-0 hgvcTd\\" tabindex=\\"0\\" type=\\"submit\\"><span class=\\"MuiButton-label\\"><span aria-hidden=\\"true\\" class=\\"fa fa-power-off\\"></span></span><span class=\\"MuiTouchRipple-root\\"></span></button></form></div></div><div class=\\"navbar-right\\"><div>LanguageToggle</div></div></nav>"`;
